components/mdns/mdns.c

```diff
@@ -185,7 +185,7 @@ static inline void _mdns_clean_netif_ptr(mdns_if_t tcpip_if)
 static mdns_if_t _mdns_get_if_from_esp_netif(esp_netif_t *esp_netif)
 {
     for (int i = 0; i < MDNS_MAX_INTERFACES; ++i) {
-        if (esp_netif == s_esp_netifs[i].netif) {
+        if (esp_netif == s_esp_netifs[i].netif || (s_esp_netifs[i].predefined && esp_netif == esp_netif_from_preset_if(s_esp_netifs[i].predef_if))) {
```

Comment:
Could you please explain the reason behind this change? I thought the static array already contains the predef netif, correct?

Reply:
The predefined netifs in the static array are NULL when firstly calling this function if IPv4 is disabled, and the `mld6_joingroup_netif` will not be called rightly then.

Reply:
Added comment here/

Reply:
> The predefined netifs in the static array are NULL 

@gabsuren Could you please check?

@wqx6 Do you use the predefined interfaces (default STA, AP or Ethernet) or do you initialize netifs manually?

Reply:
Yes, I used predefined STA interface